radix-engine/src/blueprints/package/package.rs

```diff
@@ -1145,21 +1145,19 @@ impl PackageNativePackage {
                         ),
                     },
                     function_exports,
-                    virtual_lazy_load_functions: definition_init
-                        .schema
-                        .functions
-                        .virtual_lazy_load_functions
-                        .into_iter()
-                        .map(|(key, export_name)| {
-                            (
-                                key,
+                    hook_exports: {
+                        let mut hook_exports = btreemap!();
+                        if let Some(export_name) = definition_init.schema.hooks.on_virtualize {
+                            hook_exports.insert(
+                                BlueprintHook::OnVirtualize,
                                 PackageExport {
                                     code_hash,
                                     export_name,
                                 },
-                            )
-                        })
-                        .collect(),
+                            );
+                        }
+                        hook_exports
+                    },
                 };
                 definitions.insert(blueprint.clone(), definition);
```

Comment:
Not sure if it's here but we should also be preventing Scrypto/WASM packages from registering hooks

Reply:
The same way as we prevents users from using collections.